src/components/Navbar/Language.tsx

```diff
@@ -1,33 +1,86 @@
-import React from 'react'
+import React, { useEffect, useRef, useState } from 'react'
 import { useRouter } from 'next/router'
+import Image from 'next/image'
+import LanguageItem from '../LanguageItem'
 
 const NavbarLanguage = () => {
+  const [isMenuOpen, setMenuOpen] = useState<boolean>(false)
+  const [selectedLocale, setSelectedLocale] = useState<string>('en')
+  const flagRef = useRef<HTMLDivElement>(null)
   const router = useRouter()
+  const { locales } = router
 
-  const handleLocaleClick = (locale: string) => {
+  useEffect(() => {
+    document.addEventListener('mousedown', handleMouseClick)
+
+    return () => {
+      document.removeEventListener('mousedown', handleMouseClick)
+    }
+  }, [])
+
+  const handleDropdownToggle = () => {
+    setMenuOpen((prevState) => !prevState)
+  }
```

Comment:
You can just go for one line statement. And since you shortened the word **previous**, you can go for just **state**.
```suggestion
  const handleDropdownToggle = () => setMenuOpen((state) => !state)
```